operator-common/src/main/java/io/strimzi/operator/common/model/Ca.java

```diff
@@ -386,11 +382,27 @@ private int initCaKeyGeneration(Secret caKeySecret) {
                 LOGGER.warnOp("Secret {}/{} is missing generation annotation {}",
                         caKeySecret.getMetadata().getNamespace(), caKeySecret.getMetadata().getName(), ANNO_STRIMZI_IO_CA_KEY_GENERATION);
             }
-            return Annotations.intAnnotation(caKeySecret(), ANNO_STRIMZI_IO_CA_KEY_GENERATION, INIT_GENERATION);
+            return Annotations.intAnnotation(caKeySecret, ANNO_STRIMZI_IO_CA_KEY_GENERATION, INIT_GENERATION);
         }
         return INIT_GENERATION;
     }
 
+    private Map<String, String> initCaCertData(boolean generateCa, Secret caCertSecret) {
+        if (generateCa) {
+            return caCertSecret == null ? new HashMap<>() : caCertSecret.getData();
+        } else {
+            return caCertSecret.getData();
+        }
+    }
+
+    private Map<String, String> initCaKeyData(boolean generateCa, Secret caKeySecret) {
+        if (generateCa) {
+            return caKeySecret == null ? new HashMap<>() : caKeySecret.getData();
+        } else {
+            return singletonMap(CA_KEY, caKeySecret.getData().get(CA_KEY));
```

Comment:
I know this is the same code as before but can you remember me why we are re-creating a singleton map with just the key when it's what we already have withing the corresponding Secret? (so why returning `caKeySecret.getData()` isn't enough? :thinking:

Reply:
I assume it's just because we only expect to use the entry under ca.key, so we are explicitly ignoring all the other data that might be added to the Secret accidentally. But that's just me guessing